crates/legacy/src/service.rs

```diff
@@ -180,6 +183,9 @@ pub struct GlobalState<Types: NodeType> {
 
     pub block_size_limits: BlockSizeLimits,
 
+    // A mapping from transaction hash to its status
+    pub tx_status: Arc<RwLock<lru::LruCache<Commitment<Types::Transaction>, TransactionStatus>>>,
```

Comment:
Is there a reason we want to wrap this in an `Arc`?

Reply:
Right, I wrapped it in an Arc just in case we need to use spawn with a clone of GlobalState in the future, even though we don’t have such a case yet.

Reply:
We're wrapping the whole GlobalState in an Arc already in ProxyGlobalState, so that seems superfluous

Reply:
Updated in [ea9e606](https://github.com/EspressoSystems/marketplace-builder-core/pull/236/commits/ea9e606661f7a03164a77df21f88a20a28a2ed5d).